src/components/inputs/FormControl.js

```diff
@@ -19,69 +17,53 @@ const styles = theme => ({
         },
         zIndex: 'auto',
     },
-    formControlWithMarginTop: {
-        marginTop: theme.spacing(2),
+    errorContainer: {
+        paddingLeft: theme.spacing(2),
+        paddingTop: theme.spacing(1),
     },
     error: {
         color: theme.palette.error.main,
-    },
-    marginTopZero: {
-        marginTop: 0,
+        fontSize: 14,
+        padding: 0,
     },
 });
 
-function FormControlComponent({
-    classes,
-    children,
-    withMarginTop,
-    errors,
-    marginTopZero,
-    id,
-}) {
-    const classNames = [classes.formControl];
-    // FIXME the logic seems to inverted here, but changing it leads to lots of visual bugs in Iaso
-    if (!withMarginTop) {
-        classNames.push(classes.formControlWithMarginTop);
-    }
-
-    // placing marginTopZero here to make sure it overides whatever margin was given by withMarginTop
-    if (marginTopZero) {
-        classNames.push(classes.marginTopZero);
-    }
+function FormControlComponent({ classes, children, errors, id, hideError }) {
     const extraProps = {};
     if (id) {
         extraProps.id = id;
     }
 
     return (
         <FormControl
-            className={classNames.join(' ')}
+            className={classes.formControl}
             variant="outlined"
             {...extraProps}
         >
             {children}
-            {errors.length > 0 &&
-                errors.map(error => (
-                    <Typography key={error} className={classes.error}>
-                        {error}
-                    </Typography>
-                ))}
+            {errors.length > 0 && !hideError && (
+                <Box className={classes.errorContainer}>
+                    {errors.map(error => (
+                        <Typography key={error} className={classes.error}>
```

Comment:
The key might error if for whatever reason one or several elements of the array are `null` or `undefined`